edb/pgsql/metaschema.py

```diff
@@ -2916,13 +2925,18 @@ def _generate_database_views(schema):
     return views
 
 
-def _generate_role_views(schema):
-    Role = schema.get('sys::Role')
-    member_of = Role.getptr(schema, 'member_of')
-    bases = Role.getptr(schema, 'bases')
-    ancestors = Role.getptr(schema, 'ancestors')
-    annos = Role.getptr(schema, 'annotations')
-    int_annos = Role.getptr(schema, 'annotations__internal')
+def _generate_role_views(schema: s_schema.Schema) -> List[dbops.View]:
+    Role = schema.get('sys::Role', type=s_objtypes.ObjectType)
+    member_of = Role.getptr(schema, s_name.UnqualName('member_of'))
+    assert isinstance(member_of, s_links.Link)
```

Comment:
Would it be worth adding a `type` argument to `getptr`?

Reply:
Yeah, that'd be better.